src/core/server/http/router/request.ts

```diff
@@ -138,12 +154,19 @@ export class KibanaRequest<
       enumerable: false,
     });
 
-    this.route = deepFreeze(this.getRouteInfo());
+    this.route = deepFreeze(this.getRouteInfo(request));
     this.socket = new KibanaSocket(request.raw.req.socket);
+    this.events = this.getEvents(request);
+  }
+
+  private getEvents(request: Request): KibanaRequestEvents {
+    const end$ = fromEvent<void>(request.raw.req, 'end').pipe(first());
+    return {
+      aborted$: fromEvent<void>(request.raw.req, 'aborted').pipe(first(), takeUntil(end$)),
+    } as const;
```

Comment:
I'm wondering if we should add a`replay` effect. Is there a risk for the request to already be aborded when entering the request handler?

Reply:
That's a tricky part... We can add `replay` to be double-safe.